### DISCUSSION TITLE
Toggle from favorite to TV

### DISCUSSION THREAD

**Discussion by johnluber** [avantrec/soco-cli#24] [Q&A] [answered]
I’ve got an Arc, sub and two rears setup and working well as TV surround sound.  I’ve also setup up a Sonos favorite that points to an icecast server which plays our turntable through the same speaker setup.

I’ve scripted such that if you start a record on the turntable then the music will automatically plan on the Arc setup.  What I can’t seem to figure out is how to return the Sonos setup back to the TV.  Maybe I’m missing something but is there a way to use soco-cli to do this?

**Comment by pwt** [comment]
Hi:

I've just released v0.4.25, which adds a `switch_to_tv` action. Unfortunately, I currently have no way of testing this, so if you get a chance to try it, please let me know if it works.

**Comment by johnluber** [comment]
That's awesome.  I expect I'm probably doing something wrong but I ran:

sudo pip3 install soco-cli -U

And, got this along with a lot of other output. (Note: I did run the update a couple of times)
Requirement already up-to-date: soco-cli in /usr/local/lib/python3.8/dist-packages (0.4.25)

However, when I run:

sonos -v

I get:

soco-cli version: 0.4.23
soco version:     0.24.1
python version:   3.8.10

I also run:

sonos --check_for_update

And, get:
An update is available: v0.4.25

So, I'm not sure I'm actually on 4.25.

If it helps, from the time asked the question until now I was able to use the soco library switch_to_tv and it does seem to work based on remotely trying it.  Proof will be in front of the tv.

**Comment by pwt** [comment, marked as answer]
It looks like you've installed v0.4.25, but that's not what's being run.

It's possible that you previously installed SoCo-CLI using pip3 without `sudo`, in which case the site-packages would end up in `~..local` and take precedence (I think) over the system site-packages installed using `sudo`.

Try `pip3 install -U soco-cli` without `sudo`, then `sonos -v` again.

If SoCo's `switch_to_tv` method works, then so will SoCo-CLI's action, since that's what it uses. Thanks for confirming.

**Reply by johnluber** [reply]
Yes, that seems to have been the case.  Running the update without sudo did upgrade the version to 4.25 and a quick test monitoring the S2 app suggests that switching back and forth is working.  I just have to test once I get home and get see it live.

Is there a preferred approach (ie; sudo or not) to installing soco-cli?

**Reply by pwt** [reply]
> Is there preferred approach (ie; sudo or not) to installing soco-cli?

Nothing that's specific to SoCo-CLI -- just normal Python practices really.

- Most people that use Python extensively use multiple virtualenvs to isolate Python installations for different purposes.
- For more casual use, it's fine to do what you've done, but I'd recommend using `pip3` without `sudo`, so that installed packages get placed in your home directory in `.local`, not in the system site-packages.

**Reply by johnluber** [reply]
Ok, thank you.  I appreciate the insight and the time you've spent developing soco.
